org.eclipse.jdt.debug.tests/tests/org/eclipse/jdt/debug/tests/breakpoints/ConditionalBreakpointsWithFileClass.java

```diff
@@ -0,0 +1,57 @@
+package org.eclipse.jdt.debug.tests.breakpoints;
+
+import org.eclipse.jdt.debug.core.IJavaLineBreakpoint;
+import org.eclipse.jdt.debug.core.IJavaThread;
+import org.eclipse.jdt.debug.tests.AbstractDebugTest;
+
+public class ConditionalBreakpointsWithFileClass extends AbstractDebugTest {
+
+	public ConditionalBreakpointsWithFileClass(String name) {
+		super(name);
+	}
+
+	public void testFileConditionalBreakpointForTrue() throws Exception {
+		String typeName = "FileConditionSnippet2";
+		IJavaLineBreakpoint bp2 = createLineBreakpoint(7, typeName);
+		createConditionalLineBreakpoint(374, "java.io.File", "child.equals(\"test\")", true);
+		IJavaThread mainThread = null;
+		try { //
+			mainThread = launchToBreakpoint(typeName);
+			mainThread.getTopStackFrame().stepInto();
+			// System.out.println("--> " + mainThread.getTopStackFrame().getLineNumber());
+			Thread.sleep(100);
+			mainThread.resume();
+			Thread.sleep(100);
+			// System.out.println("--> " + mainThread.getTopStackFrame().getLineNumber());
+			int hitLine = 0;
+			if (mainThread.isSuspended()) {
+				hitLine = mainThread.getStackFrames()[0].getLineNumber();
+			}
+			assertTrue("Should've hit on line 374 but hit on " + hitLine, (hitLine == 374));
```

Comment:
Two things: why do you have `if (mainThread.isSuspended())` as condition? It should be an assert.
So you will see whether the thread is suspended or not.
Second: please instead of `assertTrue(x == y)` please use **assertEquals**.

Both will not explain the fail but would help to understand the root cause.

Reply:
Will apply these changes and check again.

Reply:
@iloveeclipse Tried out several approaches and still fails in Jenkins ->
<img width="1432" alt="image" src="https://github.com/user-attachments/assets/45ba6a56-dc8d-4c68-b6bf-625409984822">

Is it because we are invoking java inbuilt class during test ?

Reply:
@iloveeclipse Done all the changes and tested, kindly review the changes